src/typings.d.ts

```diff
@@ -0,0 +1,9 @@
+import * as _d3 from 'd3';
+
+// External libraries that are included as scripts such as D3 are declared as globals
+// here with their appropriate types. This typings files is compiled into
+// any ts_library or ng_module that requires these types and it makes them available
+// as global variables that are properly typed in typescript.
+declare global {
+  const d3: typeof _d3;
```

Comment:
I feel like this is the opposite of what we want, long-term -- we want users to import d3 if they use it, so that taze can add the proper dependency when needed.

Reply:
Alex and I both agree that we need another example with d3 imported properly. This example is the only way that works at the moment with where ts_devserver and the rollup_bundle production bundler is at right now. Getting it to work with an import will require changes to both and needs to some thought.

Reply:
@evmar my understanding is that some libraries are documented such that developers expect to import them, but some are only accessible via globals. I think ABC needs to demonstrate both ways? If we only document how to use ES imports then I think some libraries would be unusable?

Reply:
I think the typings usually match how the library works, so you only need a shim thing like here when you're using the library in a way contrary to its intended use.

Reply:
@gregmagolan I'm hesitant to merge this if the intended use of d3 is to import it.
What's a library that provides a better test case for ambient imports?

Reply:
I don't know of one off the top of my head. I'll think about it.